libtiledbsoma/test/test_indexer.py

```diff
@@ -0,0 +1,191 @@
+    context = _validate_soma_tiledb_context(SOMATileDBContext(tiledb.default_ctx()))
+    indexer = build_index(keys, context)
+    results = indexer.get_indexer(lookups)
+    panda_indexer = pd.Index(keys)
+    panda_results = panda_indexer.get_indexer(lookups)
+    assert np.equal(results.all(), panda_results.all())
+
+
+test_data = [
+    {
+        "keys": [-1, -1, -1, 0, 0, 0],
+        "lookups": [1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5],
+        "pass": False,
+    },
+    {
+        "keys": [
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+        ],
+        "lookups": [-10000, 1, 2, 3, 5, 6],
+        "pass": False,
+    },
+    {
+        "keys": [-1, 1, 2, 3, 4, 5],
+        "lookups": [
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+        ],
+        "pass": True,
+    },
+    {
+        "keys": [-10000, -100000, 200000, 5, 1, 7],
+        "lookups": [
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+        ],
+        "pass": True,
+    },
+    {
+        "keys": [-10000, -200000, 1000, 3000, 1, 2],
+        "lookups": [
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+            -1,
+            1,
+            2,
+            3,
+            4,
+            5,
+        ],
+        "pass": True,
+    },
+    {
+        "keys": [i for i in range(1, 10000)],
+        "lookups": [i for i in range(1, 10)],
+        "pass": True,
+    },
+    {
+        "keys": [1, 2, 3, 4, 5, 2**63 - 1],
+        "lookups": [i for i in range(2**63 - 1000, 2**63 - 1)],
+        "pass": True,
+    },
+]
+
+
+def test_indexer():
+    for data in test_data:
+        indexer_test(data["keys"], data["lookups"], not data["pass"])
+
+
+if __name__ == "__main__":
```

Comment:
Since you fixed this to run with Pytest now, this main function is not necessary. I do think in a future PR we should parameterize these inputs and test for exceptions using `pytest.raises` instead.

Reply:
You don't need to call test_indexer - pytest will pick up that unit test automatically